### ISSUE TITLE
[Request]: Allow offline / cached mode

### ISSUE THREAD

**Issue by CaptainMorgan12** [GeopJr/Tuba#309]
### Describe the request

Maybe out of scope due to GTK limitations but allowing for asyncronous post updates and allow for caching so when not having network connection it will still load last defined # of posts say 15.

### Implementation Details

- [X] This should be an option in settings.
- [ ] This should be only available to some fediverse backends. (Include which ones on the above field).
- [X] This is client-only (and shouldn't sync with the instance).
- [X] This follows the [GNOME HIG](https://developer.gnome.org/hig/).

**Comment by CaptainMorgan12** [comment]
![tuba](https://github.com/GeopJr/Tuba/assets/80182265/a5ee5cde-ea8b-4800-8564-cd011306ac34)

**Comment by GeopJr** [comment]
(I haven't looked into restoring from cache yet, I'll reply about it when I do)

> The easiest to resolve is pull to refresh function or having a refresh button prominently displayed.

There's pull to refresh now #155 (on main)

Going from initially disconnected to connected is a bit more complex than just refreshing however, there's many other connections that need to be done prior, from instance and account information to custom emojis and notifications. Ideally there should be a queue that gets processed when internet connection is established (or keep retrying every x seconds)

**Comment by CaptainMorgan12** [comment]
Ok nice! Since restart of the app establishes the connections again currently and reliably maybe the pull to refresh would do more in certain instances when not connected to any accounts...

**Comment by nekohayo** [comment]
I think this would also probably speed up perceived performance quite a bit on startup, even if it needs some sort of spinner/progressbar/status indication at the top (rather than fullscreened) to say that it's still refreshing things...
